src/main/java/htsjdk/samtools/reference/FastaSequenceIndexCreator.java

```diff
@@ -0,0 +1,179 @@
+     * .gzi index required for use it with samtools.
+     *
+     * @param fastaFile the FASTA file.
+     *
+     * @return a fai index.
+     *
+     * @throws SAMException for formatting errors.
+     * @throws IOException  if an IO error occurs.
+     */
+    public static FastaSequenceIndex buildFromFasta(final Path fastaFile) throws IOException {
+        try(final AsciiLineReader in = new AsciiLineReader(IOUtil.openFileForReading(fastaFile))) {
+
+            // sanity check reference format:
+            // 1. Non-empty file
+            // 2. Header name starts with >
+            String previous = in.readLine();
+            if (previous == null) {
+                throw new SAMException("Cannot index empty file: " + fastaFile);
+            } else if (previous.charAt(0) != '>') {
+                throw new SAMException("Wrong sequence header: " + previous);
+            }
+
+            // initialize the sequence index
+            int sequenceIndex = -1;
+            // the location should be kept before iterating over the rest of the lines
+            long location = in.getPosition();
+
+            // initialize an empty index and the entry builder to null
+            final FastaSequenceIndex index = new FastaSequenceIndex();
+            FaiEntryBuilder entry = null;
+
+            // read the lines two by two
+            for (String line = in.readLine(); previous != null; line = in.readLine()) {
+                // in this case, the previous line contains a header and the current line the first sequence
+                if (previous.charAt(0) == '>') {
+                    // first entry should be skipped
+                    if (entry != null) {
+                        // add the entry to the index
+                        index.add(entry.build());
+                    }
+                    // update the sequence index
+                    sequenceIndex++;
+                    entry = new FaiEntryBuilder(sequenceIndex, previous, line, location);
+                } else if (line != null && line.charAt(0) == '>') {
+                    // update the location, next iteration the sequence will be handled
+                    location = in.getPosition();
+                } else if (line != null && !line.isEmpty()) {
+                    // update in case it is not a blank-line
+                    entry.updateWithSequence(line);
+                }
+                // set the previous to the current line
+                previous = line;
+            }
+
+            // add the last entry
+            index.add(entry.build());
+
+            // and return the index
+            return index;
+        }
+    }
+
+    // utility class for building the FastaSequenceIndexEntry
+    private static class FaiEntryBuilder {
+        private final int index;
+        private final String contig;
+        private final long location;
+        // the bytes per line is the bases per line plus the end of the line
```

Comment:
I think you could probably catch this quite easily at the end for any file with multiple sequences.  You could check to ensure that the start position you have for each sequence (which is the correct physical start location) matches what you get if you calculate the start of the last sequence plus the number of bytes expected given bytesPerLine.